gradle-jdks/src/main/java/com/palantir/gradle/jdks/JdksPlugin.java

```diff
@@ -38,7 +38,9 @@ public void apply(Project rootProject) {
         JdkManager jdkManager = new JdkManager(
                 jdksExtension.getJdkStorageLocation(), jdkDistributions, new JdkDownloaders(jdksExtension));
 
-        rootProject.getPluginManager().apply(BaselineJavaVersions.class);
```

Comment:
Uh, isn't plugin application idempotent? This should be safe. Is your large monorepo doing some classpath shenanigans? Or using the `plugins {}` block widely?

Reply:
https://docs.gradle.org/current/dsl/org.gradle.api.plugins.PluginManager.html#org.gradle.api.plugins.PluginManager:apply(java.lang.Class)

> void apply(Class<?> type)
>
> Applies the given plugin. Does nothing if the plugin has already been applied.

Reply:
You could probably look at the gradle build scan to get some idea of there are multiple `BaselineJavaVersions` plugins being applied, which would imply some classloader issues. Even with this change, I would expect that it would still go wrong as there would be different versions of the extension class.